File: test_doubao_api.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import requests
import json

# 豆包AI API配置
API_URL = "https://ark.cn-beijing.volces.com/api/v3/chat/completions"
API_KEY = "e779c50a-bc8c-4673-ada3-30c4e7987018"

def test_api():
    """测试豆包API"""
    headers = {
        "Authorization": f"Bearer {API_KEY}",
        "Content-Type": "application/json"
    }
    
    data = {
        "model": "doubao-pro-32k-240515",
        "messages": [
            {
                "role": "user",
                "content": "你好，请告诉我豆包AI是否支持图像生成功能？"
            }
        ],
        "max_tokens": 100
    }
    
    try:
        print("正在测试豆包API...")
        print(f"URL: {API_URL}")
        print(f"Headers: {headers}")
        print(f"Data: {json.dumps(data, indent=2, ensure_ascii=False)}")
        
        response = requests.post(API_URL, headers=headers, json=data, timeout=30)
        print(f"\n状态码: {response.status_code}")
        print(f"响应头: {dict(response.headers)}")
        print(f"响应内容: {response.text}")
        
        if response.status_code == 200:
            result = response.json()
            print("\n✅ API调用成功!")
            print(json.dumps(result, indent=2, ensure_ascii=False))
        else:
            print(f"\n❌ API调用失败: {response.status_code}")
            
    except Exception as e:
        print(f"❌ 请求失败: {e}")

if __name__ == "__main__":
    test_api()

File: test_doubao_api.js
// 测试豆包推理模型API调用

const https = require('https');

const apiKey = 'e779c50a-bc8c-4673-ada3-30c4e7987018';
const model = 'doubao-seed-1-6-thinking-250715';

// 准备请求数据
const requestData = {
  model: model,
  messages: [
    {
      content: [
        {
          image_url: {
            url: "https://ark-project.tos-cn-beijing.ivolces.com/images/view.jpeg"
          },
          type: "image_url"
        },
        {
          text: "图片主要讲了什么?",
          type: "text"
        }
      ],
      role: "user"
    }
  ]
};

// 准备请求选项
const options = {
  hostname: 'ark.cn-beijing.volces.com',
  path: '/api/v3/chat/completions',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${apiKey}`
  }
};

// 发送请求
const req = https.request(options, (res) => {
  console.log(`状态码: ${res.statusCode}`);
  
  let data = '';
  
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    console.log('响应数据:');
    console.log(data);
    
    try {
      const jsonData = JSON.parse(data);
      if (jsonData.error) {
        console.error('API调用错误:', jsonData.error);
      } else {
        console.log('API调用成功!');
      }
    } catch (e) {
      console.error('解析响应数据失败:', e);
    }
  });
});

req.on('error', (e) => {
  console.error(`请求遇到问题: ${e.message}`);
});

// 写入请求数据
req.write(JSON.stringify(requestData));
req.end();

console.log('API请求已发送，等待响应...');